### PR TITLE
Evaluate skipped dsr-ponyfill.e2e.ts tests, housekeeping #2922

### PR DESCRIPTION
**Pull request by tatjana-gartner2**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves [#2922]

### CODE REVIEW

File: packages/components/src/components/tabs-bar/tabs-bar.spec.ts [hunk 1]
```diff
@@ -103,8 +103,30 @@ describe('this.activeTabIndexHandler()', () => {
     expect(spy).toBeCalledTimes(1);
   });
 
-  // TODO: missing
-  xit('should set this.direction', () => {});
+  it('should set this.direction to "next" if oldValue is smaller then this.internalTabIndex', () => {
```
Comment (tatjana-gartner2) [inline packages/components/src/components/tabs-bar/tabs-bar.spec.ts:106]:
```suggestion
  it('should set this.direction to "next" if oldValue is smaller than this.internalTabIndex', () => {
```

File: packages/components/src/components/tabs-bar/tabs-bar.spec.ts [hunk 2]
```diff
@@ -103,8 +103,30 @@ describe('this.activeTabIndexHandler()', () => {
...
+
+  it('should set this.direction to "next" if oldValue is undefined', () => {
+    const component = initComponent();
+
+    component.activeTabIndexHandler(0, undefined);
+
+    expect(component['direction']).toBe('next')
+  });
+
+  it('should set this.direction to "prev" if oldValue is not smaller then this.internalTabIndex', () => {
```
Comment (tatjana-gartner2) [inline packages/components/src/components/tabs-bar/tabs-bar.spec.ts:123]:
```suggestion
  it('should set this.direction to "prev" if oldValue is not smaller than this.internalTabIndex', () => {
```

File: packages/components/src/components/tabs-bar/tabs-bar.spec.ts [hunk 3]
```diff
@@ -103,8 +103,30 @@ describe('this.activeTabIndexHandler()', () => {
...
   });
 
-  // TODO: missing
-  xit('should set this.direction', () => {});
+  it('should set this.direction to "next" if oldValue is smaller than this.internalTabIndex', () => {
+    const component = initComponent();
+    component['internalTabIndex'] = 2;
+    jest.spyOn(component, 'setTabElements' as any).mockImplementation();
+
+    component.activeTabIndexHandler(0, 1);
```
Comment (tatjana-gartner2) [inline packages/components/src/components/tabs-bar/tabs-bar.spec.ts:111]:
```suggestion
    component.activeTabIndexHandler(0, 1);
    
```

File: packages/components/src/components/tabs-bar/tabs-bar.spec.ts [hunk 4]
```diff
@@ -103,8 +103,30 @@ describe('this.activeTabIndexHandler()', () => {
...
+    component.activeTabIndexHandler(0, undefined);
+
+    expect(component['direction']).toBe('next')
+  });
+
+  it('should set this.direction to "prev" if oldValue is not smaller than this.internalTabIndex', () => {
+    const component = initComponent();
+    component['internalTabIndex'] = 1;
+
+    component.activeTabIndexHandler(0, 2);
```
Comment (tatjana-gartner2) [inline packages/components/src/components/tabs-bar/tabs-bar.spec.ts:127]:
```suggestion
    component.activeTabIndexHandler(0, 2);
    
```

File: packages/components/src/utils/typography/index.ts [hunk 5]
```diff
@@ -1,2 +1,5 @@
 export * from './isSizeInherit';
 export * from './typography-align';
+export * from './typography-text-color';
+export * from './text-size';
```
Comment (denyo) [inline packages/components/src/utils/typography/index.ts:4]: Could be ordered alphabetically.

File: packages/components/src/components/display/display-utils.spec.ts [hunk 6]
```diff
@@ -1,32 +1,32 @@
 import { getDisplayTagType } from './display-utils';
-import * as hasSpecificSlottedTagUtil from './../../utils/dom/hasSpecificSlottedTag';
+import * as hasSpecificTagUtil from '../../utils/dom/hasSpecificDirectChildTag';
```
Comment (denyo) [inline packages/components/src/components/display/display-utils.spec.ts:2]: Maybe rename wildcard import for consistency.

File: packages/components/src/components/tabs-bar/tabs-bar.spec.ts [hunk 7]
```diff
@@ -103,8 +103,32 @@ describe('this.activeTabIndexHandler()', () => {
...
-  // TODO: missing
-  xit('should set this.direction', () => {});
+  it('should set this.direction to "next" if oldValue is smaller than this.internalTabIndex', () => {
+    const component = initComponent();
+    component['internalTabIndex'] = 2;
+    jest.spyOn(component, 'setTabElements' as any).mockImplementation();
+
+    component.activeTabIndexHandler(0, 1);
+    
+    expect(component['direction']).toBe('next');
```
Comment (denyo) [inline packages/components/src/components/tabs-bar/tabs-bar.spec.ts:113]: Would be good to verify that direction wasn't already `next` from the beginning.

File: packages/components/src/components/tabs-bar/tabs-bar.spec.ts [hunk 8]
```diff
@@ -103,8 +103,32 @@ describe('this.activeTabIndexHandler()', () => {
...
+    expect(component['direction']).toBe('next')
+  });
+
+  it('should set this.direction to "prev" if oldValue is not smaller than this.internalTabIndex', () => {
+    const component = initComponent();
+    component['internalTabIndex'] = 1;
+
+    component.activeTabIndexHandler(0, 2);
+    
+    expect(component['direction']).toBe('prev');
```
Comment (denyo) [inline packages/components/src/components/tabs-bar/tabs-bar.spec.ts:130]: Would be good to verify that direction wasn't already `prev` from the beginning.

File: packages/components/src/utils/dom/isShadowRootParentOfKind.ts [hunk 9]
```diff
@@ -3,5 +3,5 @@ import { getPrefixedTagNames, getTagName, paramCaseToCamelCase } from '..';
 
 export const isShadowRootParentOfKind = (element: HTMLElement, tagName: TagName): boolean => {
   const parentElement = (element.getRootNode() as ShadowRoot).host as HTMLElement;
-  return parentElement && getTagName(parentElement) === getPrefixedTagNames(element)[paramCaseToCamelCase(tagName)];
+  return !!(parentElement && getTagName(parentElement) === getPrefixedTagNames(element)[paramCaseToCamelCase(tagName)]);
```
Comment (denyo) [inline packages/components/src/utils/dom/isShadowRootParentOfKind.ts:6]: Prefixing is not covered, yet.

File: packages/components/src/utils/form/isParentFieldsetRequired.spec.ts [hunk 10]
```diff
@@ -1,50 +1,57 @@
...
-    const parent = document.createElement('div');
-    const child = document.createElement('div');
-    parent.appendChild(child);
-
-    expect(isParentFieldsetRequired(child)).toBe(false);
-  });
+import * as isParentOfKindUtils from '../dom/isParentOfKind';
+import * as isRequiredUtils from './isRequired';
+
+it('should call isParentOfKind with correct parameters', () => {
```
Comment (denyo) [inline packages/components/src/utils/form/isParentFieldsetRequired.spec.ts:5]:
```suggestion
it('should call isParentOfKind() with correct parameters', () => {
```

File: packages/components/src/utils/form/isParentFieldsetRequired.spec.ts [hunk 11]
```diff
@@ -1,50 +1,57 @@
...
+  const child = document.createElement('div');
+
+  isParentFieldsetRequired(child);
+
+  expect(spy).toBeCalledWith(child, 'p-fieldset');
+  expect(spy).toBeCalledWith(child, 'p-fieldset-wrapper');
+  expect(spy).toBeCalledTimes(2);
+});
+
+it('should call isRequired with correct parameters', () => {
```
Comment (denyo) [inline packages/components/src/utils/form/isParentFieldsetRequired.spec.ts:16]:
```suggestion
it('should call isRequired() with correct parameters', () => {
```

File: packages/components/src/utils/form/isParentFieldsetRequired.spec.ts [hunk 12]
```diff
@@ -1,50 +1,57 @@
...
+  expect(spy).toBeCalledWith(parent);
 });
 
-describe('with prefix', () => {
-  it('should return true if parent is required', () => {
-    const parent = document.createElement('prefixed-p-fieldset-wrapper') as HTMLElementWithRequiredProp;
-    const child = document.createElement('div');
-    parent.appendChild(child);
-    parent.required = true;
+it('should return true if isRequired and isParentOfKind return true', () => {
```
Comment (denyo) [inline packages/components/src/utils/form/isParentFieldsetRequired.spec.ts:27]:
```suggestion
it('should return true if isRequired() and isParentOfKind() return true', () => {
```

File: packages/components/src/utils/form/isParentFieldsetRequired.spec.ts [hunk 13]
```diff
@@ -1,50 +1,57 @@
...
+  jest.spyOn(isRequiredUtils, 'isRequired').mockReturnValueOnce(true);
+  jest.spyOn(isParentOfKindUtils, 'isParentOfKind').mockReturnValue(true);
+  const child = document.createElement('div');
 
-    expect(isParentFieldsetRequired(child)).toBe(true);
-  });
+  expect(isParentFieldsetRequired(child)).toBe(true);
+});
+
+it('should return false if isRequired returns false and isParentOfKind returns true', () => {
```
Comment (denyo) [inline packages/components/src/utils/form/isParentFieldsetRequired.spec.ts:35]:
```suggestion
it('should return false if isRequired() returns false and isParentOfKind() returns true', () => {
```

File: packages/components/src/utils/form/isParentFieldsetRequired.spec.ts [hunk 14]
```diff
@@ -1,50 +1,57 @@
...
+  jest.spyOn(isRequiredUtils, 'isRequired').mockReturnValueOnce(false);
+  jest.spyOn(isParentOfKindUtils, 'isParentOfKind').mockReturnValue(true);
+  const child = document.createElement('div');
+
+  expect(isParentFieldsetRequired(child)).toBe(false);
+});
+
+it('should return false if isRequired returns true and isParentOfKind returns false', () => {
```
Comment (denyo) [inline packages/components/src/utils/form/isParentFieldsetRequired.spec.ts:43]:
```suggestion
it('should return false if isRequired() returns true and isParentOfKind() returns false', () => {
```

File: packages/components/src/utils/form/isParentFieldsetRequired.spec.ts [hunk 15]
```diff
@@ -1,50 +1,57 @@
...
+
+  expect(isParentFieldsetRequired(child)).toBe(false);
+});
 
-  it('should return false if parent is not required', () => {
-    const parent = document.createElement('prefixed-p-fieldset-wrapper') as HTMLElementWithRequiredProp;
-    const child = document.createElement('div');
-    parent.appendChild(child);
-    parent.required = false;
+it('should return false if isRequired returns false and isParentOfKind returns false', () => {
```
Comment (denyo) [inline packages/components/src/utils/form/isParentFieldsetRequired.spec.ts:51]:
```suggestion
it('should return false if isRequired() returns false and isParentOfKind() returns false', () => {
```

File: packages/components/src/utils/form/isParentFieldsetRequired.spec.ts (hunk 12 above)
Comment (denyo) [inline packages/components/src/utils/form/isParentFieldsetRequired.spec.ts:27]: Perhaps having an `it.each` setup for the 4 scenarios would be easier to maintain an read.

File: packages/components/src/utils/log/warnIfDeprecatedPropValueIsUsed.spec.ts [hunk 16]
```diff
@@ -0,0 +1,57 @@
+import * as loggerUtils from "./logger";
+import * as helperUtils from './helper';
+import { warnIfDeprecatedPropValueIsUsed } from "./warnIfDeprecatedPropValueIsUsed";
+
+const warningMessage1 = "prop='deprecatedValue' is deprecated for component span and will be removed with next major release.";
```
Comment (denyo) [inline packages/components/src/utils/log/warnIfDeprecatedPropValueIsUsed.spec.ts:5]: Why are these defined in outer scope?

File: packages/components/src/utils/log/warnIfDeprecatedPropValueIsUsed.spec.ts [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+
+const SOME_INSTANCE_PROP_DEPRECATED = ['deprecatedValue'] as const;
+type SomeInstancePropDeprecated = typeof SOME_INSTANCE_PROP_DEPRECATED[number];
+const SOME_INSTANCE_PROP = [
+  'value',
+  ...SOME_INSTANCE_PROP_DEPRECATED,
+] as const;
+type SomeInstanceProp = typeof SOME_INSTANCE_PROP[number];
+
+it('should call getDeprecatedPropOrSlotWarningMessage() util with correct parameters when prop value is key of deprecationMap', () => {
```
Comment (denyo) [inline packages/components/src/utils/log/warnIfDeprecatedPropValueIsUsed.spec.ts:21]: Tests suggest that calling `getDeprecatedPropOrSlotWarningMessage` does not really do anything.

File: packages/components/src/components/tabs-bar/tabs-bar.spec.ts [hunk 18]
```diff
@@ -103,8 +103,38 @@ describe('this.activeTabIndexHandler()', () => {
...
+    component.activeTabIndexHandler(0, 1);
+    expect(component['direction']).toBe('next');
+  });
+
+  it('should set this.direction to "next" if oldValue is undefined', () => {
+    const component = initComponent();
+
+    component.activeTabIndexHandler(0, undefined);
+
+    expect(component['direction']).toBe('next');
```
Comment (denyo) [inline packages/components/src/components/tabs-bar/tabs-bar.spec.ts:124]: Would be good to verify that direction wasn't already next from the beginning.

File: packages/components/src/utils/form/isParentFieldsetRequired.spec.ts [hunk 19]
```diff
@@ -1,50 +1,38 @@
...
+});
 
-    expect(isParentFieldsetRequired(child)).toBe(false);
-  });
+it.each<[boolean, boolean, boolean]>([
+  [true, true, true],
+  [false, true, false],
+  [true, false, false],
+  [false, false, false],
+])('should for isRequired(): %s and isParentOfKind: %s return: %s', (isRequired, isParentOfKind, result) => {
```
Comment (denyo) [inline packages/components/src/utils/form/isParentFieldsetRequired.spec.ts:32]:
```suggestion
])('should for isRequired(): %s and isParentOfKind(): %s return: %s', (isRequired, isParentOfKind, result) => {
```

File: packages/components/src/utils/dom/hasSpecificDirectChildTag.spec.ts [hunk 20]
```diff
@@ -0,0 +1,42 @@
+import * as getDirectChildHTMLElementUtils from './getDirectChildHTMLElement';
+import { hasSpecificDirectChildTag } from './hasSpecificDirectChildTag';
+
+it('should call getDirectChildHTMLElement with correct parameters', () => {
```
Comment (denyo) [inline packages/components/src/utils/dom/hasSpecificDirectChildTag.spec.ts:4]:
```suggestion
it('should call getDirectChildHTMLElement() with correct parameters', () => {
```

**Review by denyo — APPROVED** [review]

**Pull request merged**